apps/prairielearn/src/pages/enroll/enroll.ts

```diff
@@ -66,29 +70,38 @@ router.post(
         // Enrollment limits can only be configured on enterprise instances, so
         // we'll also only check and enforce the limits on enterprise instances.
         if (isEnterprise()) {
-          const {
-            institution,
-            course_instance,
-            institution_enrollment_count,
-            course_instance_enrollment_count,
-          } = await queryRow(
-            sql.select_and_lock_enrollment_counts,
+          const { course, course_instance } = await queryRow(
+            sql.select_course_instance,
             { course_instance_id: req.body.course_instance_id },
             z.object({
-              institution: InstitutionSchema,
+              course: CourseSchema,
               course_instance: CourseInstanceSchema,
-              institution_enrollment_count: z.number(),
-              course_instance_enrollment_count: z.number(),
             }),
           );
+          const institution = await queryRow(
+            sql.select_and_lock_institution,
+            { institution_id: course.institution_id },
+            InstitutionSchema,
+          );
```

Comment:
I'm a little worried about locking (though this isn't new to this PR). I'm considering the worst case: 500-person lecture gets a PL link up on the screen during the first lecture and we get hammered with requests. Counting should generally be pretty fast, but...

I'm wondering if we can get away without locking. That is, we simply allow races and thus allow the enrollment limit to be exceeded by a few (which we realistically don't care about).

Reply:
I'd be happy to lock the course instance rather than the institution. I don't care if the institution enrollment limit is exceeded by a little bit. Realistically I don't really care whether any of these limits are slightly exceeded, as you say.

Reply:
Discussed with Matt: we're comfortable enough with potentially exceeding enrollment limits in the case of races that we're just going to remove the locking.

Reply:
Locking removed in 4df5f4c.